diff --git a/gtk/gtkimcontextwayland.c b/gtk/gtkimcontextwayland.c
index 6f26f14cea1..13d334ba47f 100644
--- a/gtk/gtkimcontextwayland.c
+++ b/gtk/gtkimcontextwayland.c
@@ -916,7 +916,7 @@ gtk_im_context_wayland_set_cursor_location (GtkIMContext *context,
     {
       context_wayland->cursor_rect = rect;
       notify_im_change (GTK_IM_CONTEXT_WAYLAND (context),
-                        ZWP_TEXT_INPUT_V3_CHANGE_CAUSE_OTHER);
+                        ZWP_TEXT_INPUT_V3_CHANGE_CAUSE_INPUT_METHOD);
     }
 }
 
