--- a/src/gallium/drivers/freedreno/a3xx/fd3_emit.c
+++ b/src/gallium/drivers/freedreno/a3xx/fd3_emit.c
@@ -72,6 +72,12 @@
 
    emit_const_asserts(v, regid, sizedwords);
 
+   /* Flush HLSQ lazy state updates before every indirect constant load.
+    * (pocketfed#80, mesa#12634)
+    */
+   OUT_PKT3(ring, CP_EVENT_WRITE, 1);
+   OUT_RING(ring, HLSQ_FLUSH);
+
    OUT_PKT3(ring, CP_LOAD_STATE, 2);
    OUT_RING(ring, CP_LOAD_STATE_0_DST_OFF(dst_off) |
                      CP_LOAD_STATE_0_STATE_SRC(SS_INDIRECT) |
